pkg/controller/eventing/inmemory/channel/reconcile.go

```diff
@@ -157,11 +157,26 @@ func (r *reconciler) reconcile(ctx context.Context, c *eventingv1alpha1.Channel)
 	}
 	c.Status.SetAddress(controller.ServiceHostName(svc.Name, svc.Namespace))
 
-	_, err = util.CreateVirtualService(ctx, r.client, c, svc)
-	if err != nil {
-		logger.Info("Error creating the Virtual Service for the Channel", zap.Error(err))
-		r.recorder.Eventf(c, corev1.EventTypeWarning, virtualServiceCreateFailed, "Failed to reconcile Virtual Service for the Channel: %v", err)
-		return err
+	if c.Spec.Provisioner.Name == asyncProvisionerName {
```

Comment:
Are both paths of this 'if' statement unit tested? If not, do so.

Reply:
I think this still isn't unit tested.

Reply:
Right, but when I look at the code, in both cases, both branches of if, we create exactly the same virtual service (unless I was reading code incorrectly), even if a different from `in-memory` provisioner name is used we overwrite it to `in-memory`.  So I am not sure what to test here, the only I could think of is `if` with a different than `in-memory` provisioner name, but that would test only golang's `if`. WDYT?

Reply:
I want the unit test to test that the created `VirtualService` has the same destination regardless of whether or not `in-memory` or `in-memory-channel` are used. 

As it is, I don't think we have any unit tests for the new `in-memory` behavior. As that is the provisioner we want going forward, I would like it tested. And once that is tested, I want to also test that `in-memory-channel` behaves correctly, namely that the created `VirtualService` points at the correct place.